R/initRepo.R

```diff
@@ -27,16 +41,55 @@ initRepo <- function(name="drat", basepath="~/git") {
 
     dir.create(dir)
     repo <- git2r::init(dir)
+    ## add github remote if available
+    # https://github.com/account_name/repo_name.git
+    # git@github.com:account_name/repo_name.git
+    remote_url <- switch(remote_type,
+       SSH = paste0("git@github.com:", github_acc, "/", name, ".git"),
+       HTTPS = paste0("https://github.com/", github_acc, "/", name, ".git")
+    )
+    if (!is.null(github_acc)) git2r::remote_add(repo, "origin", remote_url)
 
     writeLines("## Drat Repo", file.path(dir, "README.md"))
     git2r::add(repo, "README.md")
-    cmt <- git2r::commit(repo, "Initial Commit")
+    cmt <- git2r::commit(repo, "Initial Commit in master with README.md")
 
-    git2r::checkout(repo, "gh-pages", create=TRUE)
-    
+    git2r::checkout(repo, branch, create=TRUE)
+    # create source path
     dir.create(file.path(dir, "src"))
     dir.create(file.path(dir, "src", "contrib"))
-    ## create binary path as well ?
-
+    ## create binary path as well
+    # / should work in both windows and mac/linux
+    # picked from https://cran.r-project.org/bin/windows/contrib/
+    win_binary_base <- "bin/windows/contrib"
+    win_r_versions <- c("3.1", "3.2", "3.3", "3.4", "3.5")
+    win_binary_folders <- file.path(dir, win_binary_base, win_r_versions)
+    # picked from https://cran.r-project.org/bin/macosx/
+    mac_binary_bases <- list(mavericks = "bin/macosx/mavericks/contrib",
+                             contrib = "bin/macosx/contrib",
+                             el_capitan = "bin/macosx/el-capitan/contrib")
+    mac_r_versions <- list(mavericks = c("3.1", "3.2", "3.3"),
+                           contrib = c("3.0", "3.1", "3.2"),
+                           el_capitan = c("3.4", "3.5"))
+    mac_binary_folders <- lapply(names(mac_binary_bases), 
+                                 function(x) {
+                                   file.path(dir, mac_binary_bases[[x]],
+                                             mac_r_versions[[x]])
+                                 })
+    # create folders and empty package index files
+    binary_folders <- unlist(c(win_binary_folders, mac_binary_folders))
+    # give a context in case there is warning/error in folder creation
+    message("Creating binary repositories paths")
+    lapply(binary_folders, function(folder) {
+      dir.create(folder, recursive = TRUE)
+      file.create(file.path(folder, "PACKAGES"))
+    })
+    git2r::add(repo, ".")
+    git2r::commit(repo, paste0("Initial Commit in ", branch, 
+                               " with repositories paths"))
+    # push repo
+    if (push) git2r::push(repo, name = "origin", 
+                          refspec = paste0("refs/heads/", branch))
```

Comment:
Also note there have been issues with git2r::push to github, please check if this works with a stock cran install before merging in. - https://github.com/ropensci/git2r/issues/331